dbt/adapters/athena/python_submissions.py

```diff
@@ -182,27 +194,46 @@ def poll_until_execution_completion(self, calculation_execution_id: str) -> Any:
             calculation_execution_id (str): The ID of the calculation execution to poll.
 
         Returns:
-            str: The final state of the calculation execution, which can be one of "COMPLETED", "FAILED" or "CANCELLED".
+            str: The final state of the calculation execution, which can be one of "COMPLETED", "FAILED" or "CANCELED".
 
         Raises:
             DbtRuntimeError: If the calculation execution does not complete within the timeout period.
 
         """
         polling_interval = self.polling_interval
         while True:
-            execution_status = self.athena_client.get_calculation_execution_status(
+            timer = 0
+            execution_response = self.athena_client.get_calculation_execution(
                 CalculationExecutionId=calculation_execution_id
-            )["Status"]["State"]
-            if execution_status in ["FAILED", "CANCELLED"]:
+            )
+            execution_session = execution_response.get("SessionId", None)
+            execution_status = execution_response.get("Status", None)
+            execution_result = execution_response.get("Result", None)
+            execution_stderr_s3_path = ""
+            if execution_result:
+                execution_stderr_s3_path = execution_result.get("StdErrorS3Uri", None)
+
+            execution_status_state = ""
+            execution_status_reason = ""
+            if execution_status:
```

Comment:
suggestion (non-blocking): It would be good to log the various results here

Reply:
sure will check this out